src/components/CodePreview.tsx

```diff
@@ -1,18 +1,45 @@
-import { JetBrains_Mono } from "@next/font/google"
+"use client";
 
-const jetBrainsMono = JetBrains_Mono({ subsets: ['latin'] })
+import { useState } from "react";
+import { JetBrains_Mono } from "@next/font/google";
+
+import { Copy as CopyIcon } from "lucide-react";
+
+const jetBrainsMono = JetBrains_Mono({ subsets: ["latin"] });
 
 interface CodePreviewProps {
-  code: string
+  code: string;
+  raw?: string;
 }
 
-export function CodePreview({ code }: CodePreviewProps) {
+export function CodePreview({ code, raw }: CodePreviewProps) {
+  const [copy, setCopy] = useState("Copy to Clipboard");
```

Comment:
I think we should persist only a boolean inside the state instead of the full string. Something like`hasCopiedToClipboard` that can be `true` ou `false` and then, inside the `render` function, we can render different texts based on this boolean like `{ hasCopiedToClipboard ? 'Copied to Clipboard!' : 'Copy to Clipboard' }`.

Reply:
Done! ✨